Sources/Networking/InternalAPI.swift

```diff
@@ -13,29 +13,56 @@
 
 import Foundation
 
-final class InternalAPI {
+class InternalAPI {
 
     typealias ResponseHandler = (BackendError?) -> Void
 
     private let backendConfig: BackendConfiguration
-    private let callbackCache: CallbackCache<HealthOperation.Callback>
+    private let healthCallbackCache: CallbackCache<HealthOperation.Callback>
 
     init(backendConfig: BackendConfiguration) {
         self.backendConfig = backendConfig
-        self.callbackCache = .init()
+        self.healthCallbackCache = .init()
     }
 
     func healthRequest(signatureVerification: Bool, completion: @escaping ResponseHandler) {
         let factory = HealthOperation.createFactory(httpClient: self.backendConfig.httpClient,
-                                                    callbackCache: self.callbackCache,
+                                                    callbackCache: self.healthCallbackCache,
                                                     signatureVerification: signatureVerification)
 
         let callback = HealthOperation.Callback(cacheKey: factory.cacheKey, completion: completion)
-        let cacheStatus = self.callbackCache.add(callback)
+        let cacheStatus = self.healthCallbackCache.add(callback)
 
         self.backendConfig.addCacheableOperation(with: factory,
                                                  delay: .none,
                                                  cacheStatus: cacheStatus)
     }
 
+    @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+    func postPaywallEvents(events: [PaywallStoredEvent], completion: @escaping ResponseHandler) {
+        guard !events.isEmpty else {
+            self.backendConfig.operationDispatcher.dispatchOnMainThread {
+                completion(nil)
+            }
+            return
+        }
```

Comment:
why do we need main thread for _anything_ in paywall events?

Reply:
Good point. I did that out of habit from other operations, but `ResponseHandler` isn't `@MainActor` anyway.
I'll fix in the next PR.

Reply:
10da09b89b146875800bc09bec22ccc7a9462e3e